botocore/credentials.py

```diff
@@ -1321,6 +1323,18 @@ def _validate_source_profile(self, parent_profile, source_profile):
                 )
             )
 
+        profile = profiles[source_profile]
+
+        #ensure the profile has valid credential type
+        if not self._has_static_credentials(profile) and \
+           not self._has_assume_role_config_vars(profile):
```

Comment:
It would be nice to get these into a `_has_credentials` or similar function so that when we add support for other profile types we won't have to make this if statement worse.

Reply:
I'll break this out.